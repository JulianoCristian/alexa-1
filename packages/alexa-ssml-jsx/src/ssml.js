import schema from './schema/index';
import validate from './prop-types/validate';

export default function ssml(tagName, props, ...children) {
  const tag = throwIfUndefined(getTagDefinition(tagName), `Unsupported tag: "${tagName}"`);

  const hasProps = props || tag.defaultProps || children.length;
  const mergedProps = hasProps && { ...tag.defaultProps, ...props, ...(children.length && { children }) };
  const validatedProps = validate(tag, mergedProps);

  return { type: tag.type, ...(validatedProps && { props: validatedProps }) };
};

function throwIfUndefined(item, error) {
  if (!item) {
    throw new Error(error);
  } else {
    return item;
  }
}

function getTagDefinition(tag) {
  switch (typeof tag) {
  case 'string':
    return schema[tag];
  case 'function':
    return { ...tag, type: tag };
  default:
    return undefined;
  }
}
