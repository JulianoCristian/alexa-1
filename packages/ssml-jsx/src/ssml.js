import schema from './schema/index';
import PropTypes from 'prop-types';

export default function ssml(tagName, props, ...children) {
  const tag = throwIfUndefined(getTagDefinition(tagName), `Unsupported tag: "${tagName}"`);

  const hasProps = props || tag.defaultProps || children.length;
  const mergedProps = hasProps && { ...tag.defaultProps, ...props, ...(children.length && { children }) };

  PropTypes.checkPropTypes(tag.propTypes, mergedProps, 'prop', tagName);

  return { type: tag.type, ...(mergedProps && { props: mergedProps }) };
};

function throwIfUndefined(item, error) {
  if (!item) {
    throw new Error(error);
  } else {
    return item;
  }
}

function getTagDefinition(tag) {
  switch (typeof tag) {
  case 'string':
    return schema[tag];
  case 'function':
    return { ...tag, type: tag };
  default:
    return undefined;
  }
}
