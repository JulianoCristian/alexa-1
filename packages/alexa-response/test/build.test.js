import Response from '../src';
import ssml from 'ssml-jsx';

test('PlainText', () => {
  const response = Response.build({
    ask: 'Does this work?',
    reprompt: 'Well? Does it?',
    card: { title: 'Test', content: 'Testing if this thing works' }
  });

  expect(response.build()).toMatchSnapshot();
});

test('SSML', () => {
  const response = Response.build({
    ask: (<speak>Does this work?</speak>),
    reprompt: 'Well? Does it?',
    card: { title: 'Test', content: 'Testing if this thing works' }
  });

  expect(response.build()).toMatchSnapshot();
});
