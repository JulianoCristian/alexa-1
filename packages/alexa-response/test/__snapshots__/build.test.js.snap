// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainText 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "Testing if this thing works",
      "title": "Test",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "text": "Does this work?",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "Well? Does it?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;

exports[`SSML 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "Testing if this thing works",
      "title": "Test",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "ssml": "<speak>Does this work?</speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "Well? Does it?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;
